File: src/MiniERP.Application/Services/ProductService.cs
using MiniERP.Application.DTOs;
using MiniERP.Application.Contracts;
using MiniERP.Domain.Entities.Inventory;
using System.Collections.Generic;
using System.Linq;
using System.Threading.Tasks;

namespace MiniERP.Application.Services;

public class ProductService : IProductService
{
    private readonly IProductRepository _repo;
    private readonly ICodeSequenceService _codeSequenceService;

    public ProductService(IProductRepository repo, ICodeSequenceService codeSequenceService)
    {
        _repo = repo;
        _codeSequenceService = codeSequenceService;
    }

    public async Task<ProductDto> CreateAsync(ProductCreateDto dto)
    {
        var product = new Product
        {
            Code = string.Empty,
            Name = dto.Name,
            Description = dto.Description,
            Price = dto.Price,
            Cost = dto.Cost,
            Stock = dto.Stock,
            MinStock = dto.MinStock,
            Category = dto.Category,
            Unit = dto.Unit,
            CreatedAt = System.DateTime.UtcNow
        };

        product.Code = await _codeSequenceService.GenerateNextAsync("product", "PROD-");

        await _repo.AddAsync(product);

        return MapToDto(product);
    }

    public async Task DeleteAsync(int id)
    {
        var product = await _repo.GetByIdAsync(id);
        if (product == null) return;
        await _repo.DeleteAsync(product);
    }

    public async Task<IEnumerable<ProductDto>> GetAllAsync()
    {
        var list = await _repo.GetAllAsync();
        return list.Select(MapToDto);
    }

    public async Task<ProductDto?> GetByIdAsync(int id)
    {
        var p = await _repo.GetByIdAsync(id);
        return p == null ? null : MapToDto(p);
    }

    public async Task UpdateAsync(ProductUpdateDto dto)
    {
        var product = await _repo.GetByIdAsync(dto.Id);
        if (product == null) return;
        product.Name = dto.Name;
        product.Description = dto.Description;
        product.Price = dto.Price;
        product.Cost = dto.Cost;
        product.Stock = dto.Stock;
        product.MinStock = dto.MinStock;
        product.Category = dto.Category;
        product.Unit = dto.Unit;
        product.UpdatedAt = System.DateTime.UtcNow;
        await _repo.UpdateAsync(product);
    }

    private static ProductDto MapToDto(Product p)
    {
        return new ProductDto
        {
            Id = p.Id,
            Code = p.Code,
            Name = p.Name,
            Description = p.Description,
            Price = p.Price,
            Cost = p.Cost,
            Stock = p.Stock,
            MinStock = p.MinStock,
            Category = p.Category,
            Unit = p.Unit,
            CreatedAt = p.CreatedAt,
            UpdatedAt = p.UpdatedAt
        };
    }
}

File: MiniERP/src/MiniERP.Web/src/services/productService.ts
import api from './api';

export interface ProductCreate {
  code?: string | null;
  name: string;
  description?: string | null;
  price: number;
  cost: number;
  stock: number;
  minStock: number;
  category?: string | null;
  unit?: string | null;
}

export interface Product {
  id: number;
  code: string;
  name: string;
  description?: string | null;
  price: number;
  cost: number;
  stock: number;
  minStock: number;
  category?: string | null;
  unit?: string | null;
  createdAt: string;
  updatedAt?: string | null;
}

export const getProducts = async (): Promise<Product[]> => {
  const res = await api.get('/products');
  return res.data;
};

export const getProduct = async (id: number): Promise<Product> => {
  const res = await api.get(`/products/${id}`);
  return res.data;
};

export const createProduct = async (payload: ProductCreate): Promise<Product> => {
  const res = await api.post('/products', payload);
  return res.data;
};

export const updateProduct = async (id: number, payload: ProductCreate) => {
  await api.put(`/products/${id}`, { ...payload, id });
};

export const deleteProduct = async (id: number) => {
  await api.delete(`/products/${id}`);
};

export default { getProducts, getProduct, createProduct, updateProduct, deleteProduct };
